packages/flutter/test/material/theme_test.dart

```diff
@@ -265,4 +265,40 @@ void main() {
     expect(glyphText.text.style.color, Colors.orange);
     expect(glyphText.text.style.fontSize, 20.0);
   });
+
+  testWidgets(
+    'Same ThemeData reapplied does not trigger didChangeDepedencies',
+    (WidgetTester tester) async {
+      testBuildCalled = 0;
+      final ThemeData themeData = new ThemeData(primaryColor: const Color(0xFF000000));
+
+      await tester.pumpWidget(
+        new MaterialApp(
+          theme: themeData,
+          home: new Test(),
+        ),
+      );
+      expect(testBuildCalled, 1);
+      tester.element(find.byType(MaterialApp)).markNeedsBuild();
```

Comment:
woohoo, that was exactly the advice I was looking for. Was reading the Element code but couldn't figure out when rebuilds are triggered. Thanks!

Confirmed your suggested test fails at the previous commit point